python/src/cloudstorage/cloudstorage_api.py

```diff
@@ -273,6 +275,185 @@ def listbucket(path_prefix, marker=None, prefix=None, max_keys=None,
   return _Bucket(api, bucket, options)
 
 
+# pylint: disable=too-many-locals, too-many-branches, too-many-statements
+def compose(list_of_files, destination_file,
+            content_type=None, retry_params=None, _account_id=None):
+  '''
+    Runs the GCS Compose on the inputed files.
+    Merges between 2 and 32 files into one file. If any of the previous files had the merge
+    run on them it counts towards the 32 limit.
+  Args:
+    list_of_files: list of file name strings or dictionaries with the following format:
+      {'file_name': REQUIRED name of the file to be merged. Do not include the bucket name,
+       'Generation': OPTIONAL Used to specify what version of a file to use,
+       'IfGenerationMatch': OPTIONAL Used to fail requests if versions don't match}
+    destination_file: Path to the desired output file. Must have the bucket in the path.
+    content_type: Used to specify the content-header of the output.
```

Comment:
It defaults to "binary/octet-stream" which is odd imo. I can easily remove the functionality.